### DISCUSSION TITLE
v0.50.1 release

### DISCUSSION THREAD

**Discussion by zadam** [TriliumNext/Trilium#2609] [Releases]
* brand new external REST API called [ETAPI](https://github.com/zadam/trilium/wiki/ETAPI)
* removed username from authentication, only password is now needed
* removed setting password in the initial setup to speed up the process and avoid confusing users, it's possible to set it up later
* added possibility to reset password (thus losing protected notes, if any) from Options.
* as a result of above, typical "new document" setup is just two-click process without having to enter username & password
* optimized sync protocol which can now in some cases halve the sync/time
* unified Jump-To and quick search behavior with regards to hoisting and opening in a new tab, #2483
* search immediately when user chooses "full search" from Jump To dialog, #2483
* make whole text/code detail clickable to improve UX, add placeholder to empty code notes
* print should include promoted attributes, #2550
* add #shareRoot label to define an "index" note, closes #2567
* initial theme is chosen based on OS settings, #2516
* focus promoted/owned attributes input after activating the ribbon tab, closes https://github.com/zadam/trilium/issues/2606

<hr /><em>This discussion was created from the release <a href='https://github.com/zadam/trilium/releases/tag/v0.50.1'>v0.50.1 release</a>.</em>

**Comment by thfrei** [comment]
Thank you for your update.

@zadam I am wondering, why did you develop the new ETAPI? I like it, I mean it's good to have a new tool ;-) But what was your thinking behind it? Do you have any immediate applications or examples for it?

**Reply by zadam** [reply]
There's a relatively common use case that you want to create some notes from some external system. Various automation/reporting/integration or e.g. a custom conversion of notes from a different note taking solution. There has been also a request for command line creation of notes which isn't directly fulfilled by REST API, but can be done with a simple bash/curl script now with ETAPI.

**Comment by Nriver** [comment]
FYI, https://editor.swagger.io/ is a nice online tool to read and test the ETAPI document.
